c10/core/TensorImpl.cpp

```diff
@@ -29,27 +29,19 @@ const char * const TensorImpl::err_msg_tensor_metadata_change_not_allowed =
     "        x.set_(y)";
 
 at::Tensor& TensorImpl::grad() {
-  if (type_set_.has(TensorTypeId::VariableTensorId)) {
-    if (!autograd_meta_) autograd_meta_ = impl::GetAutogradMetaFactory()->make();
-    return autograd_meta_->grad();
-  } else {
-    AT_ERROR("grad is not implemented for Tensor");
-  }
+  if (!autograd_meta_) autograd_meta_ = impl::GetAutogradMetaFactory()->make();
```

Comment:
why does this not check the excluded TLS?

Reply:
https://github.com/pytorch/pytorch/pull/28620#discussion_r342234685